app/api/resources/user.py

```diff
@@ -498,3 +498,111 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"forgot password")
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data:
```

Comment:
we can directly check for email in data here rather than in `validate_forgot_password_email` since we are returning the same msg for both :sweat_smile:

Reply:
Yeah you are right @Tlazypanda thanks for this suggestion, this will surely reduce the redundancy in my code, I just forgot to use it in this way [as code was quite huge]!! will correct soon!!